zerver/tests/test_upload.py

```diff
@@ -91,6 +91,47 @@ def test_no_file_upload_failure(self):
         result = self.client.post("/json/set_avatar")
         self.assert_json_error(result, "You must upload exactly one avatar.")
 
+    correct_files = [
+        ('img.png', 'png_resized.png'),
+        ('img.gif', 'gif_resized.png'),
+        ('img.tif', 'tif_resized.png')
+    ]
+    corrupt_files = ['text.txt', 'corrupt.png', 'corrupt.gif']
+
+    def test_vaild_avatars(self):
+        """
+        A call to /json/set_avatar with a valid file should return a url and actually create an avatar.
+        """
+        for fname, rfname in self.correct_files:
+            # with self.subTest(fname=fname):
```

Comment:
`unittest.subTest` was introduced in Python 3.4. I have put this comment here in case someone looks at this years later, when we drop support for Python 2.